### PR TITLE
Copter: raise EKF failure even if USB is connected

### PR DESCRIPTION
**Pull request by peterbarker**
This will let EKF go bad if your PixHawk is connected to your laptop.
This doesn't seem to be a problem for the other vehicles.

This also allows the EKF to go bad in-flight if you happen to have
connected (against AP's recommendations) your companion computer to your
flight controller via USB.  Since people do this, it is better to have
the checks than not.

### CODE REVIEW

**Comment by rmackay9** [comment]
This doesn't affect the LEDs getting lighter or dimmer though right?

**Comment by peterbarker** [comment]
@rmackay9 The Notify library looks at the usb connection state via hal.gpio, so no.

File: ArduCopter/ekf_check.cpp [hunk 1]
```diff
@@ -35,7 +35,7 @@ void Copter::ekf_check()
     }
 
     // return immediately if motors are not armed, ekf check is disabled, not using ekf or usb is connected
-    if (!motors->armed() || ap.usb_connected || (g.fs_ekf_thresh <= 0.0f)) {
+    if (!motors->armed() || (g.fs_ekf_thresh <= 0.0f)) {
```
Comment (WickedShell) [inline ArduCopter/ekf_check.cpp]: Comment above should be updated since the usb connected check is being removed.

Comment (peterbarker) [reply ArduCopter/ekf_check.cpp]: Fixed.  It was already stuffed - I just stuffed it further...

File: ArduCopter/Copter.h [hunk 2]
```diff
@@ -316,7 +316,7 @@ class Copter : public AP_HAL::HAL::Callbacks {
             uint8_t logging_started         : 1; // 6       // true if dataflash logging has started
             uint8_t land_complete           : 1; // 7       // true if we have detected a landing
             uint8_t new_radio_frame         : 1; // 8       // Set true if we have new PWM data to act on from the Radio
-            uint8_t usb_connected           : 1; // 9       // true if APM is powered from USB connection
+            uint8_t usb_connected_unused    : 1; // 9       // UNUSED
```
Comment (WickedShell) [inline ArduCopter/Copter.h]: Why can't unused values be deleted from this?

**Comment by peterbarker** [comment]
On Mon, 25 Jun 2018, WickedShell wrote:

> In ArduCopter/Copter.h:
> 
> > @@ -316,7 +316,7 @@ class Copter : public AP_HAL::HAL::Callbacks {
>              uint8_t logging_started         : 1; // 6       // true if dataflash logging has started
>              uint8_t land_complete           : 1; // 7       // true if we have detected a landing
>              uint8_t new_radio_frame         : 1; // 8       // Set true if we have new PWM data to act on from the Radio
> -            uint8_t usb_connected           : 1; // 9       // true if APM is powered from USB connection
> +            uint8_t usb_connected_unused    : 1; // 9       // UNUSED
> 
> Why can't unused values be deleted from this?

These are logged directly into the dataflash.  I've assumed that we should 
try to minimise changes in here to aid analysis.

**Pull request merged**